olp-cpp-sdk-core/src/client/repository/ApiCacheRepository.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ApiCacheRepository.h"
+
+#include <olp/core/cache/KeyValueCache.h>
+#include <olp/core/logging/Log.h>
+
+namespace {
+constexpr auto kLogTag = "ApiCacheRepository";
+constexpr time_t kLookupApiExpiryTime = 3600;
+
+std::string CreateKey(const std::string& hrn, const std::string& service,
+                      const std::string& serviceVersion) {
+  return hrn + "::" + service + "::" + serviceVersion + "::api";
+}
+}  // namespace
+
+namespace olp {
+namespace repository {
+
+ApiCacheRepository::ApiCacheRepository(
+    const client::HRN& hrn, std::shared_ptr<cache::KeyValueCache> cache)
+    : hrn_(hrn), cache_(cache) {}
+
+void ApiCacheRepository::Put(const std::string& service,
+                             const std::string& version,
+                             const std::string& url) {
+  std::string hrn(hrn_.ToCatalogHRNString());
```

Comment:
If we only use HRN as string, why are we saving the HRN as HRN in the first place?